### PR TITLE
Fix data theme for gathercmsdata

### PR DESCRIPTION
**Pull request by joyet-simon**
### Reason for change

[Linear task](https://linear.app/almapay/issue/ECOM-2272/[-ps]-fix-data-theme-for-gathercmsdata)
For GatherCmsData, the CMS info "themes" data that we send to the Alma API is split to "theme_name" and "theme_version".

### Code changes

Split the "themes" data to "theme_name" and "theme_version", update unit tests and make compatible with Prestashop version under 1.7

### How to test

_As a reviewer, you are encouraged to test the PR locally._

On a Prestashop environnement in version under 1.7, call the URL to collect CMS data and check in the result if you have correct data.
You can also run unit test

### Checklist for authors and reviewers

- [ ] The title of the PR uses business wording, not technical jargon, for the changelog readers to understand it
- [ ] The PR implements the changes asked in the referenced task / issue
- [ ] The automated tests are compliant with the [testing strategy](https://www.notion.so/almapay/Backend-testing-strategy-06c642cec1bf47b9b8feca3a91ea8d4a)
- [ ] The tests are relevant, and cover the corner/error cases, not only the happy path
- [ ] You understand the impact of this PR on existing code/features
- [ ] The changes include adequate logging and Datadog traces
- [ ] Documentation is updated (API, developer documentation, ADR, Notion...)

### CODE REVIEW

**Review by Francois-Gomis — APPROVED** [review]

File: alma/lib/Helpers/ThemeHelper.php [hunk 1]
```diff
@@ -54,21 +54,36 @@ public function __construct()
...
                 $themeConfig = \Symfony\Component\Yaml\Yaml::parseFile($themeConfigPath);
                 $themeVersion = $themeConfig['version'] ?: 'undefined';
-                $themeName = $themeName . ' ' . $themeVersion;
+            }
+        } else {
+            $themeDirectory = _PS_THEME_DIR_;
+            $configFilePath = $themeDirectory . 'config.xml';
+
+            if (file_exists($configFilePath)) {
+                $xmlContent = simplexml_load_file($configFilePath);
```
Comment (webaaz) [inline alma/lib/Helpers/ThemeHelper.php:80]: We need XML extension. is this already mandatory for Prestashop?

File: alma/lib/Proxy/index.php [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+header('Expires: Mon, 26 Jul 1997 05:00:00 GMT');
+header('Last-Modified: ' . gmdate('D, d M Y H:i:s') . ' GMT');
+
+header('Cache-Control: no-store, no-cache, must-revalidate');
+header('Cache-Control: post-check=0, pre-check=0', false);
+header('Pragma: no-cache');
+
+header('Location: ../');
+exit;
```
Comment (webaaz) [inline alma/lib/Proxy/index.php:11]: New line missing

File: alma/tests/Unit/Proxy/index.php [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+header('Expires: Mon, 26 Jul 1997 05:00:00 GMT');
+header('Last-Modified: ' . gmdate('D, d M Y H:i:s') . ' GMT');
+
+header('Cache-Control: no-store, no-cache, must-revalidate');
+header('Cache-Control: post-check=0, pre-check=0', false);
+header('Pragma: no-cache');
+
+header('Location: ../');
+exit;
```
Comment (webaaz) [inline alma/tests/Unit/Proxy/index.php:11]: Same here

File: alma/tests/Unit/Validators/index.php [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+header('Expires: Mon, 26 Jul 1997 05:00:00 GMT');
+header('Last-Modified: ' . gmdate('D, d M Y H:i:s') . ' GMT');
+
+header('Cache-Control: no-store, no-cache, must-revalidate');
+header('Cache-Control: post-check=0, pre-check=0', false);
+header('Pragma: no-cache');
+
+header('Location: ../');
+exit;
```
Comment (webaaz) [inline alma/tests/Unit/Validators/index.php:11]: same here...

**Review by webaaz — CHANGES_REQUESTED** [review]

File: alma/lib/Helpers/ThemeHelper.php (hunk 1 above)
Comment (joyet-simon) [reply alma/lib/Helpers/ThemeHelper.php:80]: simplexml_load_file is a native PHP function so we don't need a XML extension I think

File: alma/lib/Builders/Validators/index.php [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+header('Expires: Mon, 26 Jul 1997 05:00:00 GMT');
+header('Last-Modified: ' . gmdate('D, d M Y H:i:s') . ' GMT');
+
+header('Cache-Control: no-store, no-cache, must-revalidate');
+header('Cache-Control: post-check=0, pre-check=0', false);
+header('Pragma: no-cache');
+
+header('Location: ../');
+exit;
```
[the remark this answers is not in the document]
Comment (joyet-simon) [reply alma/lib/Builders/Validators/index.php:11]: The new line is present, Github doesn't show it

File: alma/tests/Unit/Proxy/index.php (hunk 3 above)
Comment (joyet-simon) [reply alma/tests/Unit/Proxy/index.php:11]: The new line is present, Github doesn't show it

File: alma/tests/Unit/Validators/index.php (hunk 4 above)
Comment (joyet-simon) [reply alma/tests/Unit/Validators/index.php:11]: The new line is present, Github doesn't show it

File: alma/lib/Proxy/index.php (hunk 2 above)
Comment (joyet-simon) [reply alma/lib/Proxy/index.php:11]: The new line is present, Github doesn't show it

File: alma/lib/Helpers/ThemeHelper.php (hunk 1 above)
Comment (joyet-simon) [reply alma/lib/Helpers/ThemeHelper.php:80]: The extension "ext-simplexml" is include in Prestashop (1.6 and 1.7)

**Review by webaaz — APPROVED** [review]

**Pull request merged**